airbyte-commons/src/main/java/io/airbyte/commons/logging/MaskedDataInterceptor.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.logging;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.yaml.Yamls;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.rewrite.RewritePolicy;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.apache.logging.log4j.message.SimpleMessage;
+import org.apache.logging.log4j.status.StatusLogger;
+
+/**
+ * Custom Log4j2 {@link RewritePolicy} used to intercept all log messages and mask any JSON
+ * properties in the message that match the list of maskable properties.
+ * <p>
+ * The maskable properties file is generated by a Gradle task in the {@code :airbyte-config:specs}
+ * project. The file is named {@code specs_secrets_mask.yaml} and is located in the
+ * {@code src/main/resources/seed} directory of the {@link :airbyte-config:init} project.
+ */
+@Plugin(name = "MaskedDataInterceptor",
+        category = "Core",
+        elementType = "rewritePolicy",
+        printObject = true)
+public class MaskedDataInterceptor implements RewritePolicy {
+
+  protected static final Logger logger = StatusLogger.getLogger();
+
+  /**
+   * Mask string used as a replacement value for any sensitive properties found in the log message.
+   */
+  public static final String MASK = "********";
```

Comment:
We have a mask declared here `airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/JsonSecretsProcessor.java`. Would it make sense to make sure that we keep this mask standard across the codebase? i.e. all of it points to one constant?

Reply:
@cgardens Totally make sense.  However, I think it we require some refactoring.  `airbyte-commons`, where the rewrite policy lives does not depend on `airbyte-config:config-persistence`, where the `JsonSecretsProcessor` lives.  In addition, the mask file generator is in `airbyte-config:specs` also does not depend on `airbyte-config:config-persistence`, though it does currently use `airbyte-commons`.  My suggestion would be to move the mask and `airbyte_secret` constant to something in `airbyte-commons` to avoid the dependency entanglement that will be caused by just making those constants in `JsonSecretsProcessor` public.  Let me know if you think this approach is sensible or if there are other options.

Reply:
@cgardens I went ahead and did the refactor proposed above (move constants to `airbyte-commons`, refactor all places that use them to now use the shared constants).

Reply:
super! thanks.